### PR TITLE
Refactor some UI tests by introducing a bunch of robots

### PR DESCRIPTION
**Pull request by alexandrepiveteau**

### CODE REVIEW

**Review by matt989253 — APPROVED** [review]

**Review by matt989253 — APPROVED** [review]

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt [hunk 1]
```diff
@@ -67,89 +68,247 @@ data class Facades(
...
+    override val facades: Facades,
+    override val infrastructure: Infrastructure,
+    override val strings: LocalizedStrings,
+    override val user: AuthenticationUser,
+) : TestEnvironment
+
+/** An interface representing a testing environment with an authenticated user. */
+interface AuthenticatedTestEnvironment : TestEnvironment {
+  override val user: AuthenticatedUser
+  override operator fun component4(): AuthenticatedUser = user
```
Comment (Fouad-sys) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:108]: [Question] Why is this function called component4?

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/ColorChoiceTabBar.kt [hunk 2]
```diff
@@ -80,12 +81,18 @@ fun ColorChoiceTabItem(
...
       modifier =
-          modifier.let { if (selected) it.dashedBorder(borderWidth, borderColor, shape) else it }) {
+          modifier
+              .clip(shape)
+              .selectable(selected, onClick = onClick)
+              .then(
+                  if (selected) Modifier.dashedBorder(borderWidth, borderColor, shape)
+                  else Modifier,
+              ),
+  ) {
```
Comment (Fouad-sys) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/ColorChoiceTabBar.kt:91]: [Question] Why was this UI change also part of this pull request?

**Review by Fouad-sys — APPROVED** [review]
Looks great!

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/ColorChoiceTabBar.kt (hunk 2 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/ColorChoiceTabBar.kt:91]: To fix some semantics nodes issues which are required to properly implement the UI tests.

In this case, the selectableGroup Modifier was missing.

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt (hunk 1 above)
Comment (alexandrepiveteau) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:108]: It's an operator which allows for the destructuring syntax.

**Pull request merged**